### PR TITLE
Misc transformer fixes

### PR DESCRIPTION
**Pull request by hariharans29**
### Description
1. SkipLayerNormalization has a new output (https://github.com/microsoft/onnxruntime/pull/13988) and the symbolic shape inference script needs corresponding updates 

2. The greedy sampling op (https://github.com/microsoft/onnxruntime/pull/13426) shouldn't re-use the logits buffer as its corresponding kernel doesn't seem to support it yet.

### Motivation and Context
Fix some transformer issues

### CODE REVIEW

File: onnxruntime/contrib_ops/cuda/transformers/generation_device_helper.cc [hunk 1]
```diff
@@ -556,7 +556,10 @@ Status GreedySearchProcessLogits(
   // In greedy search, next_token_scores is next_token_logits.
   gsl::span<T>& next_token_scores = greedy_state->next_token_scores;
 
-  auto is_reuse_logits_buffer = (input_length == 1);
+  // TODO(hasesh/wy): Support re-using logits buffer for the sampling case.
+  // Currently, we cannot re-use the logits because the sampling logic expects
+  // `next_token_scores` to be populated.
+  auto is_reuse_logits_buffer = !do_sampling && (input_length == 1);
```
Comment (hariharans29) [inline onnxruntime/contrib_ops/cuda/transformers/generation_device_helper.cc:562]: CC: @wangyems

**Review by tianleiwu — DISMISSED** [review]

**Comment by wangyems** [comment]
Thx for the fix!
One thing I'm not sure about is the exported gpt2 graph:
image
Does this look expected?

Also, the beam search fp32 model seems have parity issue with skiplayernorm fusion(if disable the skiplayernorm the parity is good). Not sure if it's just error accumulation or sth else

**Comment by hariharans29** [comment]
> Thx for the fix! One thing I'm not sure about is the exported gpt2 graph: image Does this look expected?
> 
> Also, the beam search fp32 model seems have parity issue with skiplayernorm fusion(if disable the skiplayernorm the parity is good). Not sure if it's just error accumulation or sth else

Thanks Ye for raising this ! 

1) Which portion of the exported GPT2 graph looks odd with the SLN fusion? Is it the Reshape fusion part ? Because, the only thing that strikes me as "odd" are the nodes leading upto Reshape from SLN.

2) I will debug the BeamSearch fp32 parity issue with SLN fusion

File: onnxruntime/contrib_ops/cuda/transformers/generation_device_helper.cc (hunk 1 above)
Comment (wangyems) [reply onnxruntime/contrib_ops/cuda/transformers/generation_device_helper.cc:562]: good catch!

**Review by wangyems — APPROVED** [review]
lgtm
